src/main/java/seedu/dengue/logic/commands/UndoCommand.java

```diff
@@ -0,0 +1,31 @@
+package seedu.dengue.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.dengue.logic.commands.exceptions.CommandException;
+import seedu.dengue.model.Model;
+
+/**
+ * A command that reverses a change previously made, or reverses a redo operation.
+ */
+public class UndoCommand extends Command {
+    public static final String COMMAND_WORD = "undo";
+    public static final String MESSAGE_SUCCESS = "Undo successful!";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + " performs an undo operation, where an optional argument \n"
+            + "can be provided to indicate the number of operations. \n"
+            + "eg. undo 1";
+
+    private final int numberOfUndos;
+    public UndoCommand(int numberOfUndos) {
```

Comment:
do you want to add some java docs here? same for redo

Reply:
done